features/achordion.c

```diff
@@ -131,8 +131,15 @@ bool process_achordion(uint16_t keycode, keyrecord_t* record) {
   }
 
   if (keycode == tap_hold_keycode && !record->event.pressed) {
+#ifdef RETRO_TAPPING
+    bool is_retro_key = IS_QK_MOD_TAP(keycode) || IS_QK_LAYER_TAP(keycode);
+#elif defined(RETRO_TAPPING_PER_KEY)
+    bool is_retro_key = get_retro_tapping(keycode, record);
+#else
+    bool is_retro_key = false;
+#endif
     // The active tap-hold key is being released.
-    if (achordion_state == STATE_HOLDING) {
+    if (achordion_state == STATE_HOLDING || is_retro_key) {
```

Comment:
This is clever! It's a simpler change than I first expected.

I think the idea with the change on this line is that:

* QMK core has determined the key as held, but Achordion is unsettled.
* From QMK core's point of view, the key is held and no other keys have been pressed.

So by simply plumbing the hold release event, the subsequent retro tap handling in this condition will tap the tapping keycode. Is that accurate?

Reply:
Yep this is accurate, and your explanation with the two bullet points is actually the core idea of the newer implementation. This was more specific for retro tapping only, the newer version supports all kinds of features where no other keypress happens between the press and release of the key.